packages/core/useMediaControls/index.ts

```diff
@@ -0,0 +1,481 @@
+      ({ id, label, kind, language, mode, activeCues, cues, inBandMetadataTrackDispatchType }))
+}
+
+const defaultOptions: UseMediaControlsOptions = {
+  src: '',
+  poster: '',
+  autoplay: false,
+  preload: 'auto',
+  loop: false,
+  controls: false,
+  muted: false,
+  playsinline: false,
+  autoPictureInPicture: false,
+  tracks: [],
+}
+
+export function useMediaControls(target: MaybeRef<HTMLMediaElement | null | undefined>, options: Partial<UseMediaControlsOptions>) {
+  options = { ...defaultOptions, ...options }
+
+  const currentTime = ref(0)
+  const duration = ref(0)
+  const seeking = ref(false)
+  const buffering = ref(false)
+  const volume = ref(1)
+  const waiting = ref(false)
+  const ended = ref(false)
+  const playing = ref(false)
+  const rate = ref(1)
+  const stalled = ref(false)
+  const buffered = ref<[number, number][]>([])
+  const tracks = ref<UseMediaTextTrack[]>([])
+  const selectedTrack = ref<number>(-1)
+  const isPictureInPicture = ref(false)
+
+  const supportsPictureInPicture = 'pictureInPictureEnabled' in document
+
+  /**
+   * Enables the specified track and disables the
+   * other tracks unless otherwise specified
+   *
+   * @param track The track of the id of the track to enable
+   * @param disableTracks Disable all other tracks
+   */
+  const enableTrack = (track: number | UseMediaTextTrack, disableTracks = true) => {
+    usingElRef<HTMLMediaElement>(target, (el) => {
+      const id = isNumber(track) ? track : track.id
+
+      if (disableTracks)
+        disableTrack()
+
+      el.textTracks[id].mode = 'showing'
+      selectedTrack.value = id
+    })
+  }
+
+  /**
+   * Disables the specified track. If no track is specified then
+   * all tracks will be disabled
+   *
+   * @param track The id of the track to disable
+   */
+  const disableTrack = (track?: number | UseMediaTextTrack) => {
+    usingElRef<HTMLMediaElement>(target, (el) => {
+      if (track) {
+        const id = isNumber(track) ? track : track.id
+        el.textTracks[id].mode = 'disabled'
+      }
+      else {
+        for (let i = 0; i < el.textTracks.length; ++i)
+          el.textTracks[i].mode = 'disabled'
+      }
+
+      selectedTrack.value = -1
+    })
+  }
+
+  /**
+   * Toggle picture in picture mode for the player.
+   */
+  const togglePictureInPicture = () => {
+    return new Promise((resolve, reject) => {
+      usingElRef<HTMLVideoElement>(target, async(el) => {
+        if (supportsPictureInPicture) {
+          if (!isPictureInPicture.value) {
+            (el as any).requestPictureInPicture()
+              .then(resolve)
+              .catch(reject)
+          }
+          else {
+            (document as any).exitPictureInPicture()
+              .then(resolve)
+              .catch(reject)
+          }
+        }
+      })
+    })
+  }
+
+  const stop = watch([target], () => {
+    usingElRef<HTMLMediaElement>(target, (el) => {
+      stop()
+
+      // Apply Options
+      watch(
```

Comment:
Use nested watch you will need to capture the stopper for every one of them to clean up the side-effect. Which could be troublesome to handle correctly. I would suggest to use top-level `watch` and `watchEffect` for that.

For example, this could be rewritten as 

```ts
// Apply Options
watchEffect(() => {
  const el = unref(target)
  if (!el)
    return
  el.loop = unref(options.loop)!
  el.controls = unref(options.controls)!
  el.muted = unref(options.muted)!
  el.preload = unref(options.preload)!
  el.autoplay = unref(options.autoplay)!
  el.volume = unref(volume)!
  ;(el as HTMLVideoElement).playsInline = unref(options.playsinline)!
  ;(el as HTMLVideoElement).poster = unref(options.poster) || ''
})
```

And `useEventListener` accepts `MaybeRef<EventTarget>` which will do the cleanup automatically on target changes.

Reply:
Oh good point, I'll make some changes. Thanks!

Reply:
Alright most of it should be fixed now. For the tracks the event listeners actually come from the video text track property so I had to nest those listeners within a watch and then stop them manually. If you have an idea for a better way to do it let me know